encoding/encoding_test.go

```diff
@@ -728,5 +728,20 @@ func TestNonUniqueKeys(t *testing.T) {
 		}()
 		a.So(err, s.ShouldResemble, expected)
 	}
+}
+
+func TestCastType(t *testing.T) {
+	a := s.New(t)
 
+	strct := struct {
+		Int64 uint64 `test:"i64,cast=int64"`
+	}{
+		42,
+	}
+
```

Comment:
- Please test for the actual value, the result could currently still be a nice `int64` of `0`
- Please test for error cases. What if you try to convert an int to a string? What if you try to convert other types?